src/rez/cli/env.py

```diff
@@ -116,6 +116,10 @@ def setup_parser(parser, completions=False):
     parser.add_argument(
         "--no-pkg-cache", action="store_true",
         help="Disable package caching")
+    parser.add_argument(
+        "--pkg-cache-sync", action="store_true",
+        help="Disable asynchronous package caching.  "
+             "Process will block until packages are cached.")
```

Comment:
```suggestion
    parser.add_argument(
        "--pkg-cache-mode", choices=["sync", "async"],
        help="Package caching mode")
```
I feel like this would cover more cases. With --pkg-cache-sync, if a user sets package_cache_async to false in their config, it would be impossible to undo from the CLI (ignoring setting an env var).

Reply:
:+1: